src/components/Api.js

```diff
@@ -0,0 +1,86 @@
+// Token: cec7d967-b84d-4a45-bbb1-45eb7a2c7337 Group ID: group-12
+// refactor all api calls to check for response okayness
+class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+
+    return response.json();
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._checkResponse);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.